G/GTK4/build_tarballs.jl

```diff
@@ -3,43 +3,97 @@
 using BinaryBuilder
 
 name = "GTK4"
-version = v"4.14.5"
+version = v"4.18.6"
 
 # Collection of sources required to build GTK
 sources = [
-    # https://download.gnome.org/sources/gtk/
     ArchiveSource("https://download.gnome.org/sources/gtk/$(version.major).$(version.minor)/gtk-$(version).tar.xz",
-                  "5547f2b9f006b133993e070b87c17804e051efda3913feaca1108fa2be41e24d"),
-    ArchiveSource("https://sourceforge.net/projects/mingw-w64/files/mingw-w64/mingw-w64-release/mingw-w64-v10.0.0.tar.bz2",
-                  "ba6b430aed72c63a3768531f6a3ffc2b0fde2c57a3b251450dcf489a894f0894"),
+                  "e1817c650ddc3261f9a8345b3b22a26a5d80af154630dedc03cc7becefffd0fa"),
+    FileSource("https://github.com/phracker/MacOSX-SDKs/releases/download/10.15/MacOSX10.15.sdk.tar.xz",
+               "2408d07df7f324d3beea818585a6d990ba99587c218a3969f924dfcc4de93b62"),
+    FileSource("https://sourceforge.net/projects/mingw-w64/files/mingw-w64/mingw-w64-release/mingw-w64-v10.0.0.tar.bz2",
+               "ba6b430aed72c63a3768531f6a3ffc2b0fde2c57a3b251450dcf489a894f0894"),
+    DirectorySource("bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/gtk*/
+cd $WORKSPACE/srcdir/gtk*
 
 # We need to run some commands with a native Glib
 apk update
-apk add glib-dev
+apk add glib-dev py3-pip
+
+# we need a newer meson (>= 1.5.0)
+pip3 install -U meson
+
+# meson shouldn't be so opinionated (mesonbuild/meson#4542 is incomplete)
+sed -i '/Werror=unused-command-line-argument/d' /usr/lib/python3.9/site-packages/mesonbuild/compilers/mixins/clang.py
+
+# Some of our older glib versions do not support `memfd_create`. Meson
+# checks whether this functino is available, but not all uses of
+# `memfd_create` are protected by `#ifdef`, so disable some features
+# manually if not supported. Reported as
+# <https://gitlab.gnome.org/GNOME/gtk/-/issues/7620>.
+atomic_patch -p1 ../patches/memfd.patch
 
 # This is awful, I know
 ln -sf /usr/bin/glib-compile-resources ${bindir}/glib-compile-resources
 ln -sf /usr/bin/glib-compile-schemas ${bindir}/glib-compile-schemas
 ln -sf /usr/bin/gdk-pixbuf-pixdata ${bindir}/gdk-pixbuf-pixdata
-# Remove gio-2.0 pkgconfig file so that it isn't picked up by post-install script.
-rm ${prefix}/lib/pkgconfig/gio-2.0.pc
 
-FLAGS=()
-if [[ "${target}" == *-apple-* ]]; then
-    FLAGS+=(-Dx11-backend=false -Dwayland-backend=false)
-elif [[ "${target}" == *-freebsd* ]]; then
-    FLAGS+=(-Dwayland-backend=false)
-elif [[ "${target}" == *-mingw* ]]; then
-    cd $WORKSPACE/srcdir/mingw*/mingw-w64-headers
+if [[ "${target}" == x86_64-linux-musl* ]]; then
+    # For some reason, all sorts of standard library paths are in `LD_LIBRARY_PATH`.
+    # On x86_64-linux-musl, this confuses the linker, and it really likes to link in libraries from these standard paths.
```

Comment:
I'm also confused by this comment. LD_LIBRARY_PATH shouldn't affect the directories searched by the linker to find libraries, that's a variable only for the dynamic loader.

Reply:
Well, seems to work now without changing `LD_LIBRARY_PATH`.